### PR TITLE
Add MiniMax-M2, Mistral4, and GLM-DSA GGUF graphs

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary
- add exact explicit-float MiniMax-M2 import with non-square Q/O geometry, full-vector Q/K RMSNorm, partial NeoX RoPE, F32 correction-biased sigmoid routing, and dynamic/static KV coverage
- add Mistral4 as its pinned DeepSeek-V2 MLA/MoE architecture, including a dedicated latent K-only cache and metadata-owned YaRN scaling; correct the vendored llama.cpp census that previously mislabeled it as Mistral3
- complete GLM-DSA tensor routing and split K/V-B transforms while preserving the dedicated DSA cache contract and fail-closed MTP boundary
- promote all three graph routes in the registry/census and regenerate the GGUF support documentation

This includes and supersedes the draft GLM-only work in #682.

## Post-review fixes
- canonicalize `glm_dsa` aliases for validation/task/package dispatch while retaining the raw alias prefix for metadata reads; the alias-spelled GGUF proves the exact mixed full/shared indexer schedule, dedicated default task, complete indexer initializer materialization, and executable dense-MLA logits
- consume Mistral4's serialized `rope.scaling.yarn_log_multiplier` through `yarn_apply_mscale`; an independent metadata oracle pins `0.0707`, factor `4`, qk dim `4` to `0.6028141`, and mutation to `0.1` discriminates the result
- fail closed during Mistral4 config extraction unless the multiplier is a non-boolean finite number in the inclusive semantic range `[0.0, 0.1]`; the wire endpoint is `float(np.float32(0.1))`, used for both validation and normalization
- reject NaN, `+inf`, `-inf`, negative values, and the immediate float32 successor above `0.1`; real `GGUFWriter`/`GGUFModel` round trips accept both endpoints and prove finite attention scale and rotary caches before graph execution
- call the immediate Mistral4 superclass initializer and force `GlmMoeDsaTask()` when callers omit `task`
- reply to and resolve both GitHub review threads

## Evidence and capability boundaries

Pinned semantic source: `ggml-org/llama.cpp@8d9af256337d1a501250f9bbf4c0859a654bddd6`.

All three real candidates exceed the 16 GiB bounded-artifact policy, so the PR records immutable metadata-only LFS evidence and uses reduced synthetic GGUFs for executable graph/cache parity:

| Architecture | Smallest pinned candidate | Complete bytes |
|---|---|---:|
| `minimax-m2` | `mradermacher/MiniMax-M2-i1-GGUF@2d4f9b1a86d32ce4dfc47db312c8d6fcae8d7b37` / `MiniMax-M2.i1-IQ1_S.gguf` | 46,514,882,176 |
| `mistral4` | `unsloth/Mistral-Small-4-119B-2603-GGUF@bd93c721735aa32c035c0f19e738cb3371fd56ff` / `Mistral-Small-4-119B-2603-UD-IQ1_M.gguf` | 32,306,941,632 |
| `glm-dsa` | `unsloth/GLM-5.2-GGUF@abc55e72527792c6e77069c99b4cb7de16fa9f23` / six `UD-IQ1_S` shards | 216,715,360,960 |

No GGUF payload was downloaded. Hub probes read file metadata only. Runtime support remains `DEFERRED`, and quantization-preserving import is `REJECTED`; callers must use `keep_quantized=False`. GLM routed DSA/MLA MTP, Mistral4 NextN/legacy MLA/Q-LoRA-free/nonzero-temperature layouts, and unowned GLM RoPE scaling all fail closed. There are no production model-ID or revision gates.

**Coverage waivers:** real-weight L2/L4/L5 and runtime generation evidence cannot fit the artifact budget. Synthetic tests instead execute full-logit prefill/cached decode, MiniMax static cache, Mistral latent-vs-expanded MLA parity, GLM indexer scheduling, alias-spelled weight closure, and weight application.

## Rebase reconciliation

Rebased onto #685 at `86e70a846b572a5266170420373781d2cac23450`. The combined tree preserves the Grok/GroveMoE/Hunyuan-MoE and MiniMax-M2/Mistral4/GLM-DSA cohorts in registry dispatch, tensor-contract validation, architecture fingerprint isolation, model coverage, parity exclusions, census, and generated docs. Combined pins are 105 supported architectures and `107` dependency-blocked / `100` evidence-only / `47` immediately implementable remaining routes.

## Validation
- focused combined GGUF/model coverage suite: `2417 passed, 273 skipped`
- immutable Hub metadata probes: `3 passed`
- broad serial non-integration suite: `8981 passed, 64 skipped, 12 deselected, 1 subtest passed`
- `lintrunner f --output oneline --all-files` and `lintrunner -a --output oneline`
- `python scripts/generate_gguf_support_docs.py --check`
- `git diff --check origin/main...HEAD`
- changed-line mypy: `0` errors (`29` pre-existing production-file errors outside changed lines)
- independent max-effort review found no significant issues and verified both cohorts, the float32 YaRN endpoint, pre-graph failure timing, fingerprint isolation, tensor mapping, cache semantics, census/docs, and fail-closed ordering

Base: `86e70a846b572a5266170420373781d2cac23450`
Head: `208f2cb3c7c5215ea9bcf8547a54109acf768bee`

### CODE REVIEW

File: src/mobius/integrations/gguf/_builder.py [hunk 1]
```diff
@@ -7098,6 +7111,33 @@ def build_from_gguf(
...
+        from mobius.tasks import GlmMoeDsaTask
+
+        if static_cache:
+            raise ValueError(
+                "static_cache=True is not supported for GLM-DSA GGUF; "
+                "dsa_kv_cache_specs() describes a per-layer-varying packed dynamic cache"
+            )
+        if task is not None and task != "glm-moe-dsa" and not isinstance(task, GlmMoeDsaTask):
+            raise ValueError("glm-dsa GGUF only supports the dedicated 'glm-moe-dsa' task")
+    if gguf_arch == "mistral4":
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/integrations/gguf/_builder.py]: Fixed. The canonical `glm-dsa` guard now assigns `GlmMoeDsaTask()` when `task` is omitted, so default dispatch cannot drift to the model class's text-generation task. The alias regression also asserts the captured import route names `GlmMoeDsaTask`, alongside mixed-indexer schedule, complete initializer materialization, and executable dense-MLA closure.

File: src/mobius/models/gguf_mistral4.py [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+                hidden_states,
+                attention_bias,
+                position_embeddings,
+                past_key_value,
+            )
+            present_key_values.append(present)
+        return self.norm(op, hidden_states), present_key_values
+
+
+class Mistral4GGUFCausalLMModel(DeepSeekV3CausalLMModel):
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/gguf_mistral4.py:291]: Fixed. `Mistral4GGUFCausalLMModel.__init__` now calls `super().__init__(base_config)` so the immediate `DeepSeekV3CausalLMModel` initializer owns inherited setup, then restores the exact GGUF config and replaces the temporary text model with `Mistral4TextModel`. The focused graph/cache/parity tests pass with this path.
